worker/uniter/relation/statemanager_test.go

```diff
@@ -169,11 +173,48 @@ func (s *stateManagerSuite) TestRemoveFailHasMembers(c *gc.C) {
 	stateTwo := relation.State{RelationId: 99}
 	stateTwo.Members = map[string]int64{"foo/1": 0}
 	s.expectState(c, stateTwo)
+	s.mockUnitGetter.EXPECT().Unit(names.NewUnitTag("foo/1")).Return(nil, nil)
 
-	mgr, err := relation.NewStateManager(s.mockUnitRW)
+	mgr, err := relation.NewStateManager(s.mockUnitRW, loggo.GetLogger("test"))
+	c.Assert(err, jc.ErrorIsNil)
+	err = mgr.RemoveRelation(99, s.mockUnitGetter, map[string]bool{})
+	c.Assert(err, gc.ErrorMatches, `*has members: \[foo/1\]`)
+}
+
+func (s *stateManagerSuite) TestRemoveIgnoresMissingUnits(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+	stateTwo := relation.State{RelationId: 99}
+	stateTwo.Members = map[string]int64{"foo/1": 0}
+	s.expectState(c, stateTwo)
+	s.expectSetStateEmpty(c)
+	s.mockUnitGetter.EXPECT().Unit(names.NewUnitTag("foo/1")).Return(nil, &params.Error{Code: "not found"})
+
+	mgr, err := relation.NewStateManager(s.mockUnitRW, loggo.GetLogger("test"))
```

Comment:
I guess we're not worried about testing that it logged a warning?

Reply:
i added an extra check